src/core/dev_api/validation_util.hpp

```diff
@@ -106,5 +105,146 @@ OPENVINO_API std::vector<PartialShape> get_node_input_partial_shapes(const ov::N
 ///
 /// \return True if rank compatible to any of others, otherwise false.
 OPENVINO_API bool is_rank_compatible_any_of(const Rank& r, std::initializer_list<Rank> others);
+
+/// \brief Infers the output batch shape for convolution forward propagation.
+///
+/// \return Infered output shape.
+OPENVINO_API PartialShape infer_convolution_forward(const Node* node,
```

Comment:
Mark as deprecated this is shape inference leftover, and should been removed soon.